shuup/core/models/_orders.py

```diff
@@ -636,11 +636,12 @@ def create_full_refund(self, restock_products=False):
         if self.has_refunds():
             raise NoRefundToCreateException
         self.cache_prices()
-        line_data = [
-            {"line": line, "quantity": line.quantity, "restock_products": restock_products}
-            for line in self.lines.all()
-            if (line.taxful_price or (line.type == OrderLineType.PRODUCT))
-        ]
+        line_data = [{
+            "line": line,
+            "quantity": line.quantity,
+            "amount": line.taxful_price.amount,
+            "restock_products": restock_products
+        } for line in self.lines.all() if (line.taxful_price or line.type == OrderLineType.PRODUCT)]
```

Comment:
For me this check seems strange `line.taxful_price or line.type == OrderLineType.PRODUCT` what's the reasoning behind it and does it actually refund every line (methods as well) even if merchants sells taxless? What's the reason for `line.taxful_price` check, seems out of place.